src/Libs/Soa/v1/Yousign.php

```diff
@@ -124,14 +124,17 @@ public function uploadDocument(string $signatureRequestId, UploadDocumentRequest
          */
         $apiClientWithAttach = clone $this->apiClient;
 
-        if (!base64_decode($uploadDocumentRequest->file_content, true)) {
-            throw new RuntimeException('File content not valid.');
+        /**
+         * Pay attention to declaration within condition.
+         */
+        if ($decoded = base64_decode($uploadDocumentRequest->file_content, true)) {
```

Comment:
non mi fa impazire l'assegnazione diretta nell'if, è una protica che si fa spesso ma no mi piace molto

Reply:
Ok rivedendo il tutto in questo caso è la strada migliore

Reply:
Eh non piace neanche a me, ed `e una cosa che nella stragrande maggioranza dei casi non va fatta, ma qui non volevo fare un decode due volte. L'ho scritto bene in grande comunque.

Reply:
PS:
> è una pratica che si fa spesso

È una pratica da evitare come la peste salvo che non ci sia un validissimo motivo per usarla.